pkg/bib/build.go

```diff
@@ -0,0 +1,205 @@
+package bib
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/podman-bootc/pkg/user"
+	"github.com/containers/podman-bootc/pkg/utils"
+
+	"github.com/containers/podman/v5/pkg/bindings/containers"
+	"github.com/containers/podman/v5/pkg/domain/entities/types"
+	"github.com/containers/podman/v5/pkg/specgen"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+const defaultBibImage = "quay.io/centos-bootc/bootc-image-builder"
+
+type BuildOption struct {
+	BibContainerImage string
+	Config            string
+	Output            string
+	Filesystem        string
+	Format            string
+	Arch              string
+	BibExtraArgs      []string
+}
+
+func Build(ctx context.Context, user user.User, imageNameOrId string, quiet bool, buildOption BuildOption) error {
```

Comment:
I think we should error out early here when --output or --config haven't been set.

For the config, we could consider creating an empty temp file and pass that on?